### DISCUSSION TITLE
Tax_fix no non-unknown values

### DISCUSSION THREAD

**Discussion by l-gallucci** [david-barnett/microViz#157] [Q&A]
Hi! Thank you for developing such a nice package for additional phyloseq integration and nice plotting.

I'm struggling with the function tax_fix as I have a dataset (from which I removed Mitochondria, Chloroplast and what else different from Bacteria).

Now, after a subset to the desired experiment group of samples...I have two ASV that are correspondent to NA and if I try tax_fix as follow:

```
env_sub %>%
 tax_fix(
  min_length = 3,
  unknowns =,
  sep = " ", anon_unique = FALSE,
  suffix_rank = "current"
 )

```

I got:

```
Row named: ASV_9595
contains no non-unknown values, returning:
'unclassified Kingdom' for all replaced levels.
Consider editing this tax_table entry manually.

Row named: ASV_9605
contains no non-unknown values, returning:
'unclassified Kingdom' for all replaced levels.
Consider editing this tax_table entry manually.
```

I was also asking to myself if there is the possibility to use nested legends for the barplot...there is a function included for that?

**Comment by david-barnett** [comment]
Hi, glad you're finding microViz useful :)

1) If you want to keep the unclassified ASVs separate you can set anon_unique to TRUE.
2) Nested legends are not currently possible in microViz, but I have spotted a package called fantaxic that offers this specific functionality - I have not had an opportunity to test it out myself though
